Exercise_1.java

```diff
@@ -1,9 +1,37 @@
-class BinarySearch { 
+public class BinarySearch { 
     // Returns index of x if it is present in arr[l.. r], else return -1 
     int binarySearch(int arr[], int l, int r, int x) 
     { 
         //Write your code here
-    } 
+        //Doing iterative binary search
+        // Time complexity O(logn)  Space Complexity O(1)
+         l = 0;
+         r = arr.length - 1;

```

Comment:
good comments, Try to use good naming conventions for variables. Helps during actual interview